presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -42,15 +44,18 @@
     /**
      * Class constructor specifying the number of bits each value will take up for encoding and decoding.
      */
-    public ZOrder(List<Integer> encodingBits)
+    public ZOrder(List<Integer> encodingBits, boolean onlyPositive)
     {
         requireNonNull(encodingBits, "Encoding bits list should not be null.");
         checkArgument(!encodingBits.isEmpty(), "Encoding bits list should not be empty.");
 
         this.encodingBits = encodingBits;
+        this.onlyPositive = onlyPositive;
 
-        totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum() + encodingBits.size();
-        maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max().getAsInt() + 1;
+        int totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum();
+        this.totalBitLength = onlyPositive ? totalBitLength : totalBitLength + encodingBits.size();
+        int maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max().getAsInt();
+        this.maxBitLength = onlyPositive ? maxBitLength : maxBitLength + 1;
```

Comment:
Can we do something as follows? 

this.totalBitLength = ...;
this.maxBitLength = ...;
if (!onlyPositive) {
    this.totalBitLength += encodingBits.size();
    this.maxBitLength += 1;
}

Reply:
Both totalBitLength and maxBitLength are final variables, so I can't modify them after their first instantiation. That is why I initially created the local variables to store the positive values.